SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -40,6 +40,13 @@ FLocalDeploymentManager::FLocalDeploymentManager()
 {
 }
 
+FLocalDeploymentManager::~FLocalDeploymentManager()
+{
+	// The idea behind this lock is to try and make the thread calling the destructor wait for the thread that is cleaning up processes
+	// if something tries to run on this object after the destructor then we're ruined anyway
+	FScopeLock StoppingDeploymentScoped(&StoppingDeployment);
```

Comment:
Looks good for stopping the issue. We should perhaps consider actually stopping the `RuntimeProcess` and `RuntimeLogFileHandle` and destructing them properly too. As well as maybe adding the scope lock to the `RuntimeProcess->OnOutput()` lambda to prevent issues with output occurring when stopping

Reply:
out of interest, for `RuntimeProcess->OnOutput().BindLambda`, what thread is the lambda actually run on? Would it be the thread that binds the lambda? My gut feeling is we will obtain the lock when gracefully shutting down the runtime, and then when the runtime outputs shutdown logs, inside the lambda we will wait for the lock to be released. Then, when the lock is released there the runtime will be ded and there is the possibility the main unreal thread has already destroyed `LocalDeploymentManager`

Reply:
good point on force stopping the RuntimeProcess, I've added `RuntimeProcess->Stop();` to the destructor

Reply:
interested to hear if I'm understanding cpp correctly - wouldn't TUniquePtr<IFileHandle> being a class member mean that when the destructor is called, the destructor of tuniqueptr and hence ifilehandle are called automatically?

Reply:
oh so i've now removed `RuntimeProcess->Stop();` from the destructor as I didn't notice that is also a toptional member

seems like the destructor of the process object also stops the process so no need to add those